ShoppingCart.java

```diff
@@ -0,0 +1,232 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ShoppingCart {
+
+	/*
+	 * Object of class Item(pencil)
+	 */
+	static Item pencil = new Item("pencil", 5.0);
+	static Item eraser = new Item("eraser", 3.5);
+	static Item sharpner = new Item("sharpner", 6.9);
+	static Item scale = new Item("scale", 25.0);
+	static Item notebook = new Item("notebook", 55.5);
+
+	
+	/*
+	 * ArrayLists are for the storing of values
+	 */
+	static ArrayList<Item> cart = new ArrayList<Item>();
+
+	
+	/**
+	 * add method is used to add item to the cart.
+	 * 
+	 * @param {string}item -item is the object of class Item
+	 * @param {integer}quantity
+	 */
+	public static void add(Item item, int quantity) {
+		if (cart.contains(item)) {
+			System.out.println("This Item already exists");
+
+		} else {
+
+			if (quantity > 0) {
+				cart.add(item);
+				item.product_quantity = quantity;
+				System.out.println("Item Added succesfully\n");
+			}
+		}
+	}
+
+	
+	/**
+	 * update method is used to update item to the cart.
+	 * 
+	 * @param {string}item -item is the object of class Item
+	 * @param {integer}quantity
+	 */
+	public static void update(Item item, int quantity) {
+
+		if (cart.contains(item)) {
+			/*
+			 * setting quantity as negative or zero will remove the product from
+			 * the cart . 
+			 */
+			if (quantity <= 0) {
+				cart.remove(item);
+				System.out.println("Item removed succesfully\n");
+			} else {
+				item.product_quantity = quantity;
+				System.out.println("Item updated succesfully\n");
+			}
+
+		}
+	}
+
+	
+	/**
+	 * showCart method is used to DISPLAY items of the cart.
+	 */
+	public static void showCart() {
+		
+		System.out.println("\nItem Name\t" + "Item Qty\t"+ "Item price(/piece)");
+		System.out.println("==========\t" + "========\t" + "==================");
+		for (Item value : cart) {

```

Comment:
A variable name should be meaningful. E.g. `value` can be  `cartItem`

Reply:
variable name changed